heureka/ui/src/lib/slices/createFiltersSlice.js

```diff
@@ -0,0 +1,227 @@
+        set(
+          produce((state) => {
+            if (!state.filters.activeFilters[entityName]) {
+              state.filters.activeFilters[entityName] = {}
+            }
+
+            if (!state.filters.activeFilters[entityName][filterLabel]) {
+              state.filters.activeFilters[entityName][filterLabel] = []
+            }
+
+            // Add the filter values and ensure uniqueness
+            state.filters.activeFilters[entityName][filterLabel] = [
+              ...new Set([
+                ...state.filters.activeFilters[entityName][filterLabel],
+                ...filterValues,
+              ]),
+            ]
+          }),
+          false,
+          "filters.addActiveFilters"
+        )
+      },
+
+      removeActiveFilter: (entityName, filterLabel, filterValue) => {
+        set(
+          produce((state) => {
+            const updatedFilters = state.filters.activeFilters[entityName][
+              filterLabel
+            ].filter((value) => value !== filterValue)
+
+            if (updatedFilters.length === 0) {
+              delete state.filters.activeFilters[entityName][filterLabel]
+            } else {
+              state.filters.activeFilters[entityName][filterLabel] =
+                updatedFilters
+            }
+          }),
+          false,
+          "filters.removeActiveFilter"
+        )
+      },
+
+      setActivePredefinedFilter: (entityName, filterName) => {
+        set(
+          produce((state) => {
+            state.filters.activePredefinedFilter[entityName] = filterName
+          }),
+          false,
+          "filters.setActivePredefinedFilter"
+        )
+      },
+
+      clearActivePredefinedFilter: (entityName) => {
+        set(
+          produce((state) => {
+            state.filters.activePredefinedFilter[entityName] = null
+          }),
+          false,
+          "filters.clearActivePredefinedFilter"
+        )
+      },
+
+      togglePredefinedFilter: (entityName, filterName) => {
+        set(
+          produce((state) => {
+            if (
+              state.filters.activePredefinedFilter[entityName] === filterName
+            ) {
+              state.filters.activePredefinedFilter[entityName] = null
+            } else {
+              state.filters.activePredefinedFilter[entityName] = filterName
+            }
+          }),
+          false,
+          "filters.togglePredefinedFilter"
+        )
+      },
+
+      loadFilterLabelValues: (entityName, filterLabel) => {
+        set(
+          produce((state) => {
+            if (!state.filters.filterLabelValues[entityName]) {
+              state.filters.filterLabelValues[entityName] = {}
+            }
+            state.filters.filterLabelValues[entityName][filterLabel] = {
+              isLoading: true,
+            }
+          }),
+          false,
+          "filters.loadFilterLabelValues.isLoading"
+        )
+
+        // Simulate a data fetch and update the state
+        set(
```

Comment:
what does this do exactly? Which part is simulated?

Reply:
Left over from the first impl of filter for services tab. I deleted it.